packages/trader/src/AppV2/Components/StopLoss/stop-loss.tsx

```diff
@@ -5,7 +5,13 @@ import { observer } from '@deriv/stores';
 import useContractDetails from 'AppV2/Hooks/useContractDetails';
 import RiskManagementItem from '../RiskManagementItem';
 
-const StopLoss = observer(() => {
+type StopLossProps = {
+    validation_params?: {
+        [key: string]: { min: number; max: number };
+    };
+};
+
+const StopLoss = observer(({ validation_params }: StopLossProps) => {
     const { contract_info } = useContractDetails();
```

Comment:
I noticed, that TP and Sl components have a lot in common:
```
const StopLoss = observer(() => {
    const { contract_info } = useContractDetails();
    const { limit_order } = contract_info;
    const { isStopLossVisible } = getContractDetailsConfig(contract_info.contract_type ?? '');

    return isStopLossVisible ? (
        <RiskManagementItem
            label={<Localize i18n_default_text='Stop loss' />}
            modal_body_content={
                <Localize i18n_default_text='When your loss reaches or exceeds the set amount, your trade will be closed automatically.' />
            }
            value={limit_order?.stop_loss?.order_amount}
            type='stop_loss'
        />
    ) : null;
});
```
And 
```
const TakeProfit = observer(() => {
    const { contract_info } = useContractDetails();
    const { limit_order } = contract_info;
    const { isTakeProfitVisible } = getContractDetailsConfig(contract_info.contract_type ?? '');
    return isTakeProfitVisible ? (
        <RiskManagementItem
            label={<Localize i18n_default_text='Take profit' />}
            modal_body_content={
                <Localize i18n_default_text='When your profit reaches or exceeds the set amount, your trade will be closed automatically.' />
            }
            value={limit_order?.take_profit?.order_amount}
            type='take_profit'
        />
    ) : null;
});
```
Maybe if can be refactored, for e.g. create one component which will just pass a proper content (for TP or SL)?

Reply:
I don't think it's a very good idea, it will just bloat one component with many similar conditions and potentially cause the split later if the functionality or UI differs in the future. Both these fields are sharing different business use cases.